src/pybind/mgr/dashboard_v2/run-backend-api-tests.sh

```diff
@@ -45,62 +63,76 @@ urllib3==1.22
+    deactivate
 
-git clone https://github.com/ceph/teuthology.git
+    git clone https://github.com/ceph/teuthology.git
 
-cd $CURR_DIR
-cd ../../../../build
+    cd $BUILD_DIR
 
-CEPH_MGR_PY_VERSION_MAJOR=$(get_cmake_variable MGR_PYTHON_VERSION | cut -d '.' -f1)
-if [ -n "$CEPH_MGR_PY_VERSION_MAJOR" ]; then
-    CEPH_PY_VERSION_MAJOR=${CEPH_MGR_PY_VERSION_MAJOR}
-else
-    if [ $(get_cmake_variable WITH_PYTHON2) = ON ]; then
-        CEPH_PY_VERSION_MAJOR=2
+    CEPH_MGR_PY_VERSION_MAJOR=$(get_cmake_variable MGR_PYTHON_VERSION | cut -d '.' -f1)
+    if [ -n "$CEPH_MGR_PY_VERSION_MAJOR" ]; then
+        CEPH_PY_VERSION_MAJOR=${CEPH_MGR_PY_VERSION_MAJOR}
     else
-        CEPH_PY_VERSION_MAJOR=3
+        if [ $(get_cmake_variable WITH_PYTHON2) = ON ]; then
+            CEPH_PY_VERSION_MAJOR=2
+        else
+            CEPH_PY_VERSION_MAJOR=3
+        fi
     fi
-fi
 
-export COVERAGE_ENABLED=true
-export COVERAGE_FILE=.coverage.mgr.dashboard
+    export COVERAGE_ENABLED=true
+    export COVERAGE_FILE=.coverage.mgr.dashboard
+
+    MGR=2 RGW=1 ../src/vstart.sh -n -d
+    sleep 10
+    cd $CURR_DIR
+}
 
-MGR=2 RGW=1 ../src/vstart.sh -n -d
-sleep 10
+run_teuthology_tests() {
+    cd "$BUILD_DIR"
+    source $TEMP_DIR/venv/bin/activate
 
-source $TEMP_DIR/venv/bin/activate
-BUILD_DIR=`pwd`
 
-if [ "$#" -gt 0 ]; then
-  TEST_CASES=""
-  for t in "$@"; do
-    TEST_CASES="$TESTS_CASES $t"
-  done
-else
-  TEST_CASES=`for i in \`ls $BUILD_DIR/../qa/tasks/mgr/dashboard_v2/test_*\`; do F=$(basename $i); M="${F%.*}"; echo -n " tasks.mgr.dashboard_v2.$M"; done`
-  TEST_CASES="tasks.mgr.test_dashboard_v2 $TEST_CASES"
-fi
+    if [ "$#" -gt 0 ]; then
+      TEST_CASES=""
+      for t in "$@"; do
+        TEST_CASES="$TEST_CASES $t"
+      done
+    else
+      TEST_CASES=`for i in \`ls $BUILD_DIR/../qa/tasks/mgr/dashboard_v2/test_*\`; do F=$(basename $i); M="${F%.*}"; echo -n " tasks.mgr.dashboard_v2.$M"; done`
+      TEST_CASES="tasks.mgr.test_dashboard_v2 $TEST_CASES"
+    fi
 
-export PATH=$BUILD_DIR/bin:$PATH
-export LD_LIBRARY_PATH=$BUILD_DIR/lib/cython_modules/lib.${CEPH_PY_VERSION_MAJOR}/:$BUILD_DIR/lib
-export PYTHONPATH=$TEMP_DIR/teuthology:$BUILD_DIR/../qa:$BUILD_DIR/lib/cython_modules/lib.${CEPH_PY_VERSION_MAJOR}/
-eval python ../qa/tasks/vstart_runner.py $TEST_CASES
+    export PATH=$BUILD_DIR/bin:$PATH
+    export LD_LIBRARY_PATH=$BUILD_DIR/lib/cython_modules/lib.${CEPH_PY_VERSION_MAJOR}/:$BUILD_DIR/lib
+    export PYTHONPATH=$TEMP_DIR/teuthology:$BUILD_DIR/../qa:$BUILD_DIR/lib/cython_modules/lib.${CEPH_PY_VERSION_MAJOR}/
+    eval python ../qa/tasks/vstart_runner.py $TEST_CASES
 
-deactivate
-killall ceph-mgr
-sleep 10
-../src/stop.sh
-sleep 5
+    deactivate
+    cd $CURR_DIR
+}
+
+cleanup_teuthology() {
+    cd "$BUILD_DIR"
+    killall ceph-mgr
+    sleep 10
+    ../src/stop.sh
+    sleep 5
+
+    cd $CURR_DIR
+    rm -rf $TEMP_DIR
+}
 
-cd $CURR_DIR
-rm -rf $TEMP_DIR
+# End sourced section
+return 2> /dev/null
 
+setup_teuthology
```

Comment:
can we move this up to the sourced section? I think usually everyone wants to run this at the start, so it would be one less step to run.

Reply:
done